expression/builtin_string.go

```diff
@@ -1502,33 +1502,26 @@ type charLengthFunctionClass struct {
 }
 
 func (c *charLengthFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinCharLengthSig{newBaseBuiltinFunc(args, ctx)}
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpInt, tpString)
```

Comment:
yes, currently, `verifyArgs` only check argument count, you can pull another request to refine them.